### ISSUE TITLE
Normalization of embedding

### ISSUE THREAD

**Issue by akshayv1** [KaidiXu/auto_LiRPA#12]
Hi, 
Thanks for sharing the code. I am trying out the mnist example here:https://github.com/KaidiXu/auto_LiRPA/blob/319252bbe6c51a60852687418575429caf9205ee/examples/vision/simple_verification.py#L19

and was interested in adding a l2 normalization layer before the linear layer, defined by:
```
class Normalize(nn.Module):
    def forward(self, x):
        norm = x.norm(p=2, dim=1, keepdim=True)
        x = x.div(norm)
        return x
  ```

But I guess this operation is not currently supported since I see an error:```NotImplementedError: There are unsupported operations```

Could you let me know if there is a way to overcome this?

**Comment by shizhouxing** [comment]
This can be solved by defining a custom operator (e.g., your `Normalize`) with your own bound derivation for the defined operator. I will shortly add a documentation on defining custom operators.

**Comment by shizhouxing** [comment]
Hi @akshayv1, please see https://auto-lirpa.readthedocs.io/en/latest/custom_op.html. Let us know if you have further questions. Thanks!

**Comment by akshayv1** [comment]
Thanks for providing this support. Could you please add an example (considering ReLU or Leaky ReLU) so that it becomes easier to follow?

**Comment by huanzhang12** [comment]
Thanks for reaching out! I understand the documentation for adding customized OPs is hard to follow at this time. We will provide an example for a custom operator but it can take some time (our team members are currently busy with a few projects). We should be able to make an example by next month.

**Comment by shizhouxing** [comment]
I think I can prepare an example recently. It’s a good suggestion to have an example.

**Comment by shizhouxing** [comment]
Hi @akshayv1 , we have added an example for custom op:

https://auto-lirpa.readthedocs.io/en/latest/custom_op.html
https://github.com/KaidiXu/auto_LiRPA/blob/master/examples/vision/custom_op.py

Please let us know if there is anything unclear. Thanks!
